donald_wade/week_04/tuesday/MTA.rb

```diff
@@ -0,0 +1,53 @@
+require "pp"
+
+subway = {
+  n: ["Times Square", "34th Street", "28th Street", "23rd Street", "Union Square", "8th Avenue"],
+
+  l: ["8th Avenue", "6th Avenue", "Union Square", "3rd Street", "1st Avenue"],
+
+  l6: ["Grand Central", "33rd Street", "28th Street", "23rd Street", "Union Square", "Astor Place"],
+}
+
+# pp $subway[:n]
+
+pp subway[:n].index("Union Square")
+union_index = subway[:n].index("Union Square")
+# pp union_index
+
+route_to_union_forwards = subway[:n].first(subway[:n].length-union_index)
+# pp route_to_union_forwards
+
+route_to_union_backwards = subway[:n].last(union_index)
+# pp route_to_union_backwards
+
+def get_to_station(on, off, myline, subway)
+  line = subway[myline]
+  start = line.index(on)
+  finish = line.index(off)
+  if start < finish
+    # puts start, finish
+    route = line[start..finish]
+  elsif start > finish
+    # puts finish, start
+    route = line[finish..start].reverse
+  end
+  return route
+end
+
+def get_to_with_change(on, lineon, off, lineoff, subway)
+  route = [get_to_station(on, "Union Square", lineon, subway), get_to_station("Union Square", off, lineoff, subway)]
```

Comment:
think how with array manipulation we did today in a class you can find "Union Square" station - common station for all lines, in order to make this part flexible and not hardcoded as it is now.